_posts/news/2021-08-04-announcing-scalajs-1.7.0.md

```diff
@@ -0,0 +1,135 @@
+
+We therefore recommend to try and *avoid* the need for that flag instead.
+We give several strategies on how to do so [on the Regular expressions documentation page]({{ BASE_PATH }}/doc/regular-expressions.html).
+That page also contains many more details on the new support of regular expressions.
+
+## New features
+
+### Add a configurable header comment in generated .js files
+
+Sometimes, it is desirable to add a header comment in the generated .js files.
+This is typically used for license information or any other metadata.
+While it has always been possible to post-process the generated .js files in the build, doing so came at the cost of destroying the source maps.
+
+Scala.js 1.7.0 introduces a new linker configuration, `jsHeader`, to specify a comment to insert at the top of .js files:
+
+{% highlight scala %}
+scalaJSLinkerConfig ~= {
+  _.withJSHeader(
+    """
+      |/* This is the header, which source maps
+      | * take into account.
+      | */
+    """.stripMargin.trim() + "\n"
+  )
+},
+{% endhighlight %}
+
+The `jsHeader` must be a combination of valid JavaScript whitespace and/or comments, and must not contain any newline character other than `\n` (the UNIX newline).
+If non-empty, it must end with a new line.
+These restrictions ensure that this feature is not abused to inject arbitrary JavaScript code in the .js file generated by the compiler, potentially compromising the compiler abstractions.
+
+## Miscellaneous
+
+### New JDK APIs
+
+The following JDK classes have been added
+
+* `java.util.concurrent.atomic.LongAdder`
+
+### Set up the `versionScheme` of library artifacts
+
+This release configures the sbt `versionScheme` setting for the library artifacts of Scala.js (with `"semver-spec"` for the public ones).
+This will reduce spurious eviction warnings in downstream projects.
+
+### Upgrade to GCC v20210601
+
+We upgraded to the Google Closure Compiler v20210601.
+
+## Bug fixes
+
+Among others, the following bugs have been fixed in 1.7.0:
+
+* [#4507](https://github.com/scala-js/scala-js/issues/4507) 1.6.0 regression: `new mutable.WrappedArrayBuilder(classTag[Unit]).result()` throws a CCE
+* [#3953](https://github.com/scala-js/scala-js/issues/3953) fastOptJS error in scalaz 7.3 with Scala.js 1.0.0
+* [#3918](https://github.com/scala-js/scala-js/issues/3918) Mixed-in field in class inside lazy val rhs is erroneously immutable in the IR -> IR checking error
+* [#4511](https://github.com/scala-js/scala-js/issues/4511) Nested JS Class in JS Class with Scala companion crashes compiler
+* [#4465](https://github.com/scala-js/scala-js/issues/4465) Default parameters in constructors of nested JS classes cause invalid IR
+* [#4526](https://github.com/scala-js/scala-js/issues/4526) Compiler crashes on nested JS class with default constructor params with private companion
+* [#4336](https://github.com/scala-js/scala-js/issues/4336) Failing `<project>/run` can `close()` subsequent runs too early
```

Comment:
I think if this is in here, scala-js/scala-js#4469 should probably also be here.